### PR TITLE
samples: philosophers: Enable thread awareness

### PR DESCRIPTION
**Pull request by nandojve**
As requested at #37516, this enables thread awareness by default on philosophers sample.  It shows gdb debugging output using the feature as reference.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]

File: samples/philosophers/README.rst [hunk 1]
```diff
@@ -63,3 +63,80 @@ Sample Output
    Philosopher 5 [C:-2]  THINKING [ 1700 ms ]
 
 Exit QEMU by pressing :kbd:`CTRL+A` :kbd:`x`.
+
+Debug Threads
+********************
+
+The Philosopher sample by default enable CONFIG_DEBUG_THREAD_INFO.  This allows
```
Comment (erwango) [inline samples/philosophers/README.rst:70]:
```suggestion
The philosophers sample by default enables CONFIG_DEBUG_THREAD_INFO. This allows
```

**Review by erwango — DISMISSED** [review]
One nitpick, otherwise LGTM, txs!

File: samples/philosophers/README.rst [hunk 2]
```diff
@@ -63,3 +63,80 @@ Sample Output
    Philosopher 5 [C:-2]  THINKING [ 1700 ms ]
 
 Exit QEMU by pressing :kbd:`CTRL+A` :kbd:`x`.
+
+Debug Threads
+********************
```
Comment (gmarull) [inline samples/philosophers/README.rst:68]:
```suggestion
Debug Threads
*************
```

File: samples/philosophers/README.rst (hunk 1 above)
Comment (gmarull) [reply samples/philosophers/README.rst:70]:
```suggestion
The Philosopher sample by default enable :kconfig:`CONFIG_DEBUG_THREAD_INFO`.  This allows
```

**Review by dcpleung — APPROVED** [review]

**Pull request merged**